app/src/main/res/layout/view_history_list_container.xml

```diff
@@ -0,0 +1,85 @@
+        android:layout_height="wrap_content"
+        android:orientation="vertical">
+
+        <org.wikipedia.views.WikiCardView
+            android:id="@+id/wiki_card_for_search"
+            android:layout_width="match_parent"
+            android:paddingTop="16dp"
+            android:layout_marginBottom="24dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="16dp">
+
+            <include
+                android:id="@+id/search_card"
+                layout="@layout/view_search_bar"
+                android:layout_width="match_parent"
+                android:layout_height="48dp"
+                android:visibility="visible" />
+        </org.wikipedia.views.WikiCardView>
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:gravity="center_vertical">
+
+            <TextView
+                android:id="@+id/history_title"
+                android:layout_width="0dp"
+                android:layout_height="48dp"
+                android:layout_gravity="start"
+                android:layout_weight="1"
+                android:gravity="center_vertical"
+                android:lineSpacingExtra="10sp"
+                android:paddingStart="16dp"
+                android:paddingEnd="16dp"
+                android:text="@string/history_list_title"
+                android:textColor="?attr/primary_text_color"
+                android:textSize="24sp"
+                android:textStyle="bold" />
+
+            <ImageView
+                android:id="@+id/history_filter"
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:layout_marginEnd="24dp"
+                android:clickable="true"
+                android:contentDescription="@string/history_filter_list_hint"
+                android:focusable="true"
+                android:foreground="?attr/actionBarItemBackground"
+                android:longClickable="true"
+                android:scaleType="centerInside"
+                app:tint="@color/base30"
+                app:srcCompat="@drawable/ic_baseline_filter_list_24" />
+
+            <ImageView
+                android:id="@+id/history_delete"
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:layout_marginEnd="16dp"
+                android:clickable="true"
+                android:contentDescription="@string/button_clear_all_recent_searches"
+                android:focusable="true"
+                android:foreground="?attr/actionBarItemBackground"
+                android:longClickable="true"
+                android:scaleType="centerInside"
+                app:tint="@color/base30"
+                app:srcCompat="@drawable/ic_delete_white_24dp" />
+        </LinearLayout>
+
+        <androidx.recyclerview.widget.RecyclerView
```

Comment:
This `RecyclerView` is now rendering all of its contents (off screen), instead of stopping at the end of the screen.  If the user has 1000 history items, it will render all 1000 of them, grinding the app to a halt.

As we've done in the past, the proper thing to do is put the "Search" button as a different item type in the RecyclerView.